python/activator/middleware_interface.py

```diff
@@ -666,24 +670,27 @@ def _export_skymap_and_templates(self, region, filter):
         # TODO: This will be replaced by the new spatial query feature in Butler.
         template_where = " OR ".join([f"htm7 in ({range[0]}..{range[1]})"
                                       for range in lsst.sphgeom.HtmPixelization(7).interior(region).ranges()])
-        try:
-            _log.debug("Searching for templates.")
-            templates = set(_filter_datasets(
-                self.central_butler, self.butler,
-                self._get_template_types(),
-                collections=self._collection_template,
-                instrument=self.instrument.getName(),
-                skymap=self.skymap_name,
-                physical_filter=filter,
-                where=template_where,
-                findFirst=True,
-                all_callback=self._mark_dataset_usage,
-            ))
-        except _MissingDatasetError as err:
-            _log.error(err)
-            templates = set()
-        else:
-            _log.debug("Found %d new template datasets.", len(templates))
+        templates = set()
+        for template_type in self._get_template_types():
+            try:
+                _log.debug("Searching for templates %s.", template_type)
+                new_datasets = set(_filter_datasets(
+                    self.central_butler, self.butler,
+                    template_type,
+                    collections=self._collection_template,
+                    instrument=self.instrument.getName(),
+                    skymap=self.skymap_name,
+                    physical_filter=filter,
+                    where=template_where,
+                    findFirst=True,
+                    all_callback=self._mark_dataset_usage,
+                ))
+            except _MissingDatasetError as err:
+                _log.error(err)
```

Comment:
Now that I changed `_filter_datasets` back to handle multiple dataset types, I think this is fine again.  If it can't find any templates, log an error and continue. Later ApPipe will fallback to single-frame or whatever next in the config.